Sources/AlgoliaSearchClient/Helpers/DisjunctiveFacetingHelper.swift

```diff
@@ -50,7 +50,10 @@ struct DisjunctiveFacetingHelper {
     var queries = [Query]()
 
     var mainQuery = query
-    mainQuery.filters = buildFilters(excluding: .none)
+    if let mainQueryFilters = mainQuery.filters, !mainQueryFilters.isEmpty {
+      mainQuery.filters = mainQueryFilters + " AND "
+    }
+    mainQuery.filters = (mainQuery.filters ?? "") + buildFilters(excluding: .none)
```

Comment:
Hi @fluff22

Thank you for taking the time to add this behavior back to the SDK :pray:

Unfortunately, this new implementation creates an issue on my side. Indeed, when there aren’t any refinements, the buildFilters method sends back an empty string, but if you do have filters on the mainQuery you will end up with a filters string equal to “your_main_query_filter  AND “ throwing a 400 error on your side;

```
{
  "message": "filters: Unexpected token end of filter expected filter at col 195",
  "status": 400
}
```

What do you think instead of this:
```
if let mainQueryFilters = mainQuery.filters, !mainQueryFilters.isEmpty {
  mainQuery.filters = mainQueryFilters + " AND "
}
mainQuery.filters = (mainQuery.filters ?? "") + buildFilters(excluding: .none)
```

we take this approach;

```
mainQuery.filters = [
  mainQuery.filters,
  buildFilters(excluding: .none)
]
 .compactMap()
 .filter { !$0.isEmpty }
 .joined(separator: " AND ")
```

Filtering out empty strings and optionality.

Let me know your thoughts.
Thanks again

Reply:
This looks great!
Thanks for testing out this PR against your use case, it's very valuable to us.

Let me update the PR and ping you when it's done.

Reply:
I also applied your idea to the sub-queries below the main one.
Let me know how it goes!

Reply:
It does look promising on my side, I'll try to run as much test as possible to make sure everything is working well and let you know asap!

Reply:
Perfect, thanks a lot!